wgsl/index.bs

```diff
@@ -880,30 +880,35 @@ shader that goes beyond the specified limits.
   <thead>
     <tr><th>Limit<th>Minimum supported value
   </thead>
-    <tr><td>Maximum number of members in a [=structure=] type<td>16383
-    <tr><td>Maximum [=nesting depth=] of a [=composite=] type<td>255
+    <tr><td>Maximum number of members in a [=structure=] type<td>1023
+    <tr><td>Maximum [=nesting depth=] of a [=composite=] type<td>15
     <tr><td>Maximum nesting depth of brace-enclosed statements in a function<td>127
     <tr><td>Maximum number of [=formal parameter|parameters=] for a function<td>255
-    <tr><td>Maximum number of case selector values in a [=statement/switch=] statement<td>16383
-    <tr><td>Maximum [=byte-size=] of an [=array=] type instantiated in the
-            [=address spaces/function=] or [=address spaces/private=] address spaces
+    <tr><td>Maximum number of case selector values in a [=statement/switch=] statement<td>1023
+    <tr><td>Maximum combined [=byte-size=] of all [=variables=] instantiated in the
+            [=address spaces/private=] address space that are [=statically accessed=] by a single
+            [=shader=]
 
             For the purposes of this limit, [=bool=] has a size of 4 bytes.
-        <td>65535
-    <tr><td>Maximum [=byte-size=] of an [=array=] type instantiated in the
-            [=address spaces/workgroup=] address space.
+        <td>8192
+    <tr><td>Maximum combined [=byte-size=] of all [=variables=] instantiated in the
+            [=address spaces/function=] address space that are declared in a single
+            [=function/function=]
```

Comment:
Strictly speaking this means you can get different results when functions are inlined.   But the implementation has the escape hatch of issuing an uncategorized error.
And like you said this is easy for programmers to understand.